src/main/java/com/cloudbees/hudson/plugins/folder/computed/ComputedFolder.java

```diff
@@ -271,6 +281,31 @@ final void updateChildren(final TaskListener listener) throws IOException, Inter
     }
 
     /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void delete() throws IOException, InterruptedException {
+        checkPermission(DELETE);
+        FolderComputation<I> computation = getComputation();
+        Executor executor = Executor.of(computation);
+        if (executor != null) {
+            LOGGER.log(Level.INFO, "Interrupting {0} in order to delete it", this);
+            executor.interrupt(Result.ABORTED, new CauseOfInterruption.UserInterruption(User.current()));
+            // give it 15 seconds or so to respond to the interrupt
+            long expiration = System.nanoTime() + TimeUnit.SECONDS.toNanos(15);
+            while (executor.isAlive()
+                    && executor.getCurrentExecutable() == computation
+                    && expiration - System.nanoTime() > 0L) {
```

Comment:
By way of example, `System.nanoTime()` could return `Long.MAX_VALUE-5000000000L` for the first call... so `expiration` would be `Long.MAX_VALUE+10000000000L`, i.e. negative, after even 1 second then `Long.MAX_VALUE+10000000000L < Long.MAX_VALUE-4000000000L` so `expiration > System.nanoTime()` will be false and we would not wait.

By doing the subtraction comparison, we roll back around and get `Long.MAX_VALUE+10000000000L - Long.MAX_VALUE-4000000000L == 6000000000L > 0`

It would be simpler to do the waiting based on `System.currentTimeMillis()` but that is subject to adjustments so is not safe for measuring elapsed time